blockprod/src/detail/block_maker.rs

```diff
@@ -185,3 +185,460 @@ impl BlockMaker {
+    }
+
+    #[tokio::test]
+    async fn collect_transactions_collect_txs_failed() {
+        let (mut manager, chain_config, chainstate, _mempool) = setup_blockprod_test();
+
+        let mock_mempool = MempoolInterfaceMock::new();
+        *mock_mempool.collect_txs_should_error.lock().unwrap() = true;
+
+        let mock_mempool_subsystem = manager.add_subsystem_with_custom_eventloop("mock-mempool", {
+            let mock_mempool = mock_mempool.clone();
+            move |call, shutdn| async move {
+                mock_mempool.run(call, shutdn).await;
+            }
+        });
+
+        let (_tx_builder, rx_builder) = crossbeam_channel::unbounded();
+
+        tokio::spawn(async move {
+            let block_maker = BlockMaker::new(
+                chain_config,
+                chainstate.clone(),
+                mock_mempool_subsystem,
+                Default::default(),
+                Id::new(H256::random_using(&mut make_pseudo_rng())),
+                BlockHeight::one(),
+                rx_builder,
+            );
+
+            let accumulator = block_maker.collect_transactions().await;
+
+            assert!(
+                *mock_mempool.collect_txs_called.lock().unwrap(),
+                "Expected collect_tx() to be called"
+            );
+
+            assert!(
+                matches!(accumulator, Err(BlockProductionError::MempoolChannelClosed)),
+                "Expected collect_tx() to fail"
+            );
+        });
+
+        manager.main().await;
+    }
+
+    #[tokio::test]
+    async fn collect_transactions_succeeded() {
+        let (mut manager, chain_config, chainstate, _mempool) = setup_blockprod_test();
+
+        let mock_mempool = MempoolInterfaceMock::new();
+
+        let mock_mempool_subsystem = manager.add_subsystem_with_custom_eventloop("mock-mempool", {
+            let mock_mempool = mock_mempool.clone();
+            move |call, shutdn| async move {
+                mock_mempool.run(call, shutdn).await;
+            }
+        });
+
+        let (_tx_builder, rx_builder) = crossbeam_channel::unbounded();
+
+        tokio::spawn(async move {
+            let block_maker = BlockMaker::new(
+                chain_config,
+                chainstate.clone(),
+                mock_mempool_subsystem,
+                Default::default(),
+                Id::new(H256::random_using(&mut make_pseudo_rng())),
+                BlockHeight::one(),
+                rx_builder,
+            );
+
+            let accumulator = block_maker.collect_transactions().await;
+
+            assert!(
+                *mock_mempool.collect_txs_called.lock().unwrap(),
+                "Expected collect_tx() to be called"
+            );
+
+            assert!(
+                accumulator.is_ok(),
+                "Expected collect_transactions() to succeed"
+            );
+        });
+
+        manager.main().await;
+    }
+
+    //
+    // Skipping unit tests as we're about rework make_block()
```

Comment:
Is this comment relevant to this test? I don't see `ignore` flag anywhere

Reply:
Sorry, what I meant by skipping was as in me skipping implementing them.